src/rt-app.c

```diff
@@ -648,8 +648,10 @@ static void set_thread_priority(thread_data_t *data, sched_data_t *sched_data)
 				exit(EXIT_FAILURE);
 			}
 
-			ret = setpriority(PRIO_PROCESS, 0,
-					sched_data->prio);
+			param.sched_priority = sched_data->prio;
+			ret = pthread_setschedparam(pthread_self(),
+					sched_data->policy,
+					&param);
 			if (ret != 0) {
 				log_critical("[%d] setpriority"
```

Comment:
Nitpick: we should change the error message as well.

Reply:
yes you're right